scripts/test_integrity-b.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+#
+# test integrity of few basic files
```

Comment:
This needs to be wired up with `gzip_tests.sh`, otherwise it won't run in CI.

I think you're overthinking this a bit.  All we need are some quick tests in `gzip_tests.sh`.  Something like the following:

```
begin_test '-t (test) option works'                                              
good_files=(                                                                     
'H4sIAAAAAAAAA3PMSVTITVTIzi9JVABTIJ5jzpGZelwAX+86ehsAAAA='                       
'H4sIAAAAAAAAAwvJSFUoLM1MzlZIKsovz1NIy69QyCrNLShWyC9LLVIoAUrnJFZVKqTkp+txAQBqzFDrLQAAAA==')
bad_files=(                                                                      
'H4sIAO1YYmAAA3PMSVTITVTIzi9JVABTIJ5jzpGZelwAX+46ehsAAAA='                       
'H4sIAO1YYmAAA3PMSVTITVTIzi85VABTIJ5jzpGZelwAX+86ehsAAAA='                       
'H4sIAAAAAAAAA3PMSVTITVTIzi9JVABTIJ5jzpGZelwAX+86ehsBAAA='                       
'H4sIAAAAAAAAAwvJSFUoLM1MzlZIKsovz1NIy69QyCrNLShWyC9LLVIogUrnJFZVKqTkp+txAQBqzFDrLQAAAA=='
'H4sIAAAAAAAAAwvJSFUoLM1MzlZIKsovz1NIy69QyCrNLShWyC9L')                          
for contents in "${good_files[@]}"; do                                           
        echo "$contents" | base64 -d | gzip -t                                     
done                                                                             
for contents in "${bad_files[@]}"; do                                            
        echo "$contents" | base64 -d > file                                        
        assert_error '\<invalid compressed data|file corrupt|unexpected end of file\>' \
                gzip -t file                                                     
done  
```                                                                           

Note that `gzip_tests.sh` runs with bash, so it doesn't need to be POSIX compliant (it looks like you ran into problems since you used `#!/bin/sh` instead of `#!/bin/bash`).  It just needs to pass shellcheck.